dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -259,34 +259,34 @@ const bool Sedp::host_is_bigendian_(!ACE_CDR_BYTE_ORDER);
 Sedp::Sedp(const RepoId& participant_id, Spdp& owner, ACE_Thread_Mutex& lock) :
   DCPS::EndpointManager<ParticipantData_t>(participant_id, lock),
   spdp_(owner),
-  publications_writer_(
-    make_id(participant_id, ENTITYID_SEDP_BUILTIN_PUBLICATIONS_WRITER), *this),
+  publications_writer_(make_rch<Writer>(
```

Comment:
Odd that the reader was already using RCH but not the writer, though perhaps for similar reasons?

Reply:
The reader is more fundamentally driven by callbacks from the transport.  The writer has a few too -- as we've seen here.  On the writer side, even data_delivered can be done without RCH as long as remove_all_msgs is used (see other PR).

Reply:
Those 4 security attributes tests passed on my local build.